### PR TITLE
add chart lint ci workflow

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
use [CT](https://github.com/helm/chart-testing)  to validate  chart:
there are three phase:
* `ct changed`:  check the chart file change of chart on target branch
* `ct lint`: check the yaml file format of chart under charts directory
* `ct install`: install chart instant on a kind cluster

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Please correct me if not.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Poor12** [comment]
If it's ok, please ping me to review. /assign

**Comment by calvin0327** [comment]
https://github.com/helm/chart-testing/issues/444

@Poor12 our chart directory not conform to chart best practices. the correct directory shoule be `/charts/karmada/*`.
and then exec `ct list-changed` or `ct lint`  take effect.

**Comment by calvin0327** [comment]
image

this is my local test result. the branch `add-chart-lint-ci-test` and `add-chart-lint-ci` both is the same of dir `/charts/karmada/*` .

**Review by Poor12 — COMMENTED** [review]
dir should be：
charts
   - karmada/*
   - OWNERS

**Comment by Poor12** [comment]
image
What's the problem in the github action?

**Comment by cyclinder** [comment]
> image What's the problem in the github action?

@calvin0327  Maybe you should change the dir of charts firstly (Open another PR), and waiting the PR merged. and then working on this..

**Comment by calvin0327** [comment]
> dir should be： charts
> 
> * karmada/*
> * OWNERS

ok, I guess i should modify the correct directory structure firstly and then modify the pr. the target branch `master` will only take effect by exec `ct list-changed`. 😅

**Comment by calvin0327** [comment]
https://github.com/karmada-io/karmada/runs/6843463720?check_suite_focus=true

Succeed run the ci workflow of ci chart lint. @Poor12  @RainbowMango 
chart file updates must be satisfy blew rules:
* correct chart format
* add chart version(+1) every updates

File: .github/workflows/lint-chart.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+name: Chart Lint
+
+env:
+  HELM_VERSION: v3.8.1
+  KIND_VERSION: v0.12.0
+  K8S_VERSION: v1.23.4
+
+on:
+  push:
+  pull_request:
```
Comment (Poor12) [inline .github/workflows/lint-chart.yaml:11]: pull_request:
    paths:
      - "charts/karmada/**"
It means that only chartfile changed will need to run this CI workflow.

File: charts/karmada/Chart.yaml [hunk 2]
```diff
@@ -12,17 +12,21 @@ description: A Helm chart for karmada
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.0.3
+version: 0.0.4
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 appVersion: v1.1.0
+
+maintainers:
```
Comment (Poor12) [inline charts/karmada/Chart.yaml:30]: maintainers should be same with OWNERs.

**Review by Poor12 — COMMENTED** [review]
Every chart file updates will need to add version in Chart.yaml? What happened if not add?

**Comment by calvin0327** [comment]
@Poor12 thanks for review. it's done

File: .github/workflows/lint-chart.yaml (hunk 1 above)
Comment (calvin0327) [reply .github/workflows/lint-chart.yaml:11]: You’re quite right.

**Comment by Poor12** [comment]
for me /lgtm
cc @RainbowMango to check

File: .github/workflows/lint-chart.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Run chart-testing (lint)
+        run: ct lint --debug --check-version-increment=false
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.2.0
+        if: steps.list-changed.outputs.changed == 'true'
+        with:
+          wait: 120s
+          kubectl_version: ${{ env.K8S_VERSION }}
```
Comment (RainbowMango) [inline .github/workflows/lint-chart.yaml:58]: I suppose we can add `version` and `node_image` here.

Comment (RainbowMango) [reply .github/workflows/lint-chart.yaml:58]: https://github.com/helm/kind-action#inputs

Comment (calvin0327) [reply .github/workflows/lint-chart.yaml]: 👌, done

**Comment by RainbowMango** [comment]
/lgtm
/approve

Thanks.

**Pull request merged**